ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -1095,10 +1100,10 @@ impl<'a> X64CodeGen<'a> {
                 }
             }
             LocalAlloc::Register => todo!(),
-            LocalAlloc::ConstInt(c) => {
-                // FIXME: Adjust for different constant sizes. Requires storing the size in the
-                // variant.
-                dynasm!(self.asm; mov Rq(reg.code()), QWORD *c as i64);
+            LocalAlloc::ConstInt { bits, v } => {
+                // FIXME: Adjust for different constant sizes?
+                assert_eq!(*bits, 64);
```

Comment:
I think this assertion can therefore be removed too?

Reply:
At some point we'll want to handle these differently, so if we remove the assert, things get swallowed. It may be that this proves too annoying, though, and we have to get rid of it, but it passes yk's test suite so...

Reply:
Doesn't it crash programs that use constant integers that are not 64-bit? Or are all constant integers promoted to 64-bit?

Reply:
I expected it to crash in the test suite. The fact that it doesn't means that either the test suite isn't good enough or we promote things somewhere. I guess we'll find out soon enough.

Reply:
OK. Will merge once I'm done with CI fiddling.